### ISSUE TITLE
Do not unconditionally store cache locally upon entering waypoint edit mode

### ISSUE THREAD

**Issue by baiti** [cgeo/cgeo#10125]
During debugging of my Adventure Lab Cache integration I noticed that in CacheDetail view, when the Waypoints pane is shown and I tap on a waypoint, a toast shows saying "The cache has been stored locally" and then I find myself in "Edit Waypoint" mode.

I would consider it a bug, to immediately store the cache locally upon entering edit mode. I still have the option to cancel out of edit and in that case I do not want to have the cache stored locally. Only if I decide to tap the checkmark icon the cache should be stored locally and preferably only if an actual change was made.

**Comment by baiti** [comment]
Perhaps I should have stated: This bug is **not** in release 2021.02.14.

it is in my cloned fork from master, however it is also not limited to what I have done in the PR, it also affects any other Geocache from other services / connectors.

**Comment by Lineflyer** [comment]
Should also happen on release 2021.02.14 AFAIR.
But you are right: It would be better to only save it once the waypoint has been changed.

**Comment by baiti** [comment]
> Should also happen on release 2021.02.14 AFAIR.
> But you are right: It would be better to only save it once the waypoint has been changed.

Yes, it is indeed also in 2021.02.14. It just happened that in my earlier test today, the cache in question was already locally stored when I tapped into Waypoint Edit mode.
